poc/experiment.py

```diff
@@ -0,0 +1,214 @@
+from datasets.discrete_choice_datasets import TrivialDiscreteChoiceProblem
+from datasets.variable_choice_datasets import ParetoChoiceProblem
+from datasets.variable_choice_datasets import TrivialVariableChoiceProblem
+from estimators import FATEDiscreteChoiceEstimator
+from estimators import FATERankingEstimator
+from estimators import FETADiscreteChoiceEstimator
+from estimators import FETARankingEstimator
+from estimators import FETAVariableChoiceEstimator
+from metrics import instance_informedness
+from modules.utilities import PairwiseDenseUtility
+from skorch_logging import SacredLogger
+
+ex = Experiment()
+
+
+@ex.config
+def my_config():
+    estimator = None
+    estimator_params = dict()
+    estimator_type = None
+    dataset = None
+    dataset_params = dict()
+    dataset_type = None
+
+
+for (estimator, config_name, estimator_type) in [
+    (FATEDiscreteChoiceEstimator, "fate_discrete_choice_estimator", "discrete_choice"),
+    (FATERankingEstimator, "fate_ranking_estimator", "ranking"),
+    (FETADiscreteChoiceEstimator, "feta_discrete_choice_estimator", "discrete_choice"),
+    (FETARankingEstimator, "feta_ranking_estimator", "ranking"),
+    (FETAVariableChoiceEstimator, "feta_variable_choice_estimator", "variable_choice"),
+]:
+    config = {
+        "estimator": estimator,
+        "estimator_params": estimator(callbacks=[], max_epochs=int(1e9)).get_params(),
+        "estimator_type": estimator_type,
+    }
+
+    ex.add_named_config(config_name, config)
+
+
+for (dataset, config_name, dataset_type) in [
+    (
+        TrivialDiscreteChoiceProblem,
+        "trivial_discrete_choice_problem",
+        "discrete_choice",
+    ),
+    (
+        TrivialRankingProblem,
+        "trivial_ranking_problem",
+        "ranking",
+    ),
+    (
+        TrivialVariableChoiceProblem,
+        "trivial_variable_choice_problem",
+        "variable_choice",
+    ),
+    (ParetoChoiceProblem, "pareto_choice_problem", "variable_choice"),
+]:
+    config = {
+        "dataset": dataset,
+        "dataset_params": {
+            "n_features": 2,
+            "n_objects": 3,
+            "n_instances": int(1e5),
+        },
+        "dataset_type": dataset_type,
+    }
+    if dataset_type == "discrete_choice":
+        config["dataset_params"]["choice_size"] = 2
+
+    for (parameter, value) in inspect.signature(dataset).parameters.items():
+        if value.default != inspect._empty:
+            config["dataset_params"][parameter] = value.default
+    ex.add_named_config(config_name, config)
+
+
+@ex.named_config
+def learning_choice_functions_feta_pareto():
+    """A (partial) reproduction of the experimental setup of the FETA paper."""
+    # Refer to Table 2 in https://arxiv.org/abs/1901.10860 for hyperparameters
+    # and reasonable ranges.
+    estimator = FETAVariableChoiceEstimator
+    estimator_params = {
+        "max_epochs": int(1e9),  # TODO look up when to terminate in the paper
```

Comment:
@kiudee do you recall what stopping criterion you used in the experiments of ["Learning Choice Functions: Concepts and Architectures"](https://arxiv.org/abs/1901.10860)? I can't find that information in the paper. It mentions a tolerance parameter, but that is only used for PairwiseSVM. The `epochs` parameter in the current cs-ranking implementation usually defaults to `10`, which seems very small (it should be at least something in the order of `edrop` in the paper).

Was it a fixed epoch value or some sort of convergence criterion?

Reply:
As far as I remember we used 1000 epochs. It is true that we did not mention this in the paper.

Reply:
Okay, thanks! Easy to forget.